libevmasm/GasMeter.cpp

```diff
@@ -111,6 +111,7 @@ GasMeter::GasConsumption GasMeter::estimateMax(AssemblyItem const& _item, bool _
 		case Instruction::CALLDATACOPY:
 		case Instruction::CODECOPY:
 		case Instruction::RETURNDATACOPY:
+		case Instruction::MCOPY:
```

Comment:
Sounds like you're right, but it's really bad that no test caught it. So now I'm trying to add one and `libsolidity/GasMeter.cpp` looks like the right place. The weird thing is that I added your example as a new test there and it passes on the current implementation. Still looking into why.

Reply:
Turns out that these tests were passing for me locally because they only run for abicoder v1. So I re-enabled them for v2 in #14790. I only had to keep some of them disabled.

So now this is fixed and also covered with tests and the estimations seem to match evmone in all the corner cases.